src/algorithms/generate.rs

```diff
@@ -78,8 +76,9 @@ pub(crate) fn generate_multi_prime_key_with_exp<R: CryptoRngCore + ?Sized>(
         }
 
         for (i, prime) in primes.iter_mut().enumerate() {
-            *prime = rng.gen_prime(todo / (nprimes - i));
-            todo -= prime.bits();
+            let bits = (todo / (nprimes - i)) as u32;
+            *prime = generate_prime_with_rng(rng, bits, bits);
```

Comment:
The bit precision (2nd arg) here seems to be wrong ?

The tests fails because there is this behind:
`let mut random = T::random_bits_with_precision(rng, bit_length, bits_precision);`

random.bits_precision() = 128, bits_precision = 85, bit_length = 85, bits_precision = 85
random.bits_precision() = 512, bits_precision = 512, bit_length = 512, bits_precision = 512
random.bits_precision() = 64, bits_precision = 6, bit_length = 6, bits_precision = 6

85 should be 128 - 6 should be 64 hmmm